lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -250,25 +250,62 @@ class MongoClientInterface {
     }
 
     getBucketAndObject(bucketName, objName, params, log, cb) {
-        this.getBucketAttributes(bucketName, log, (err, bucket) => {
+        if (params && params.versionId) {
+            // eslint-disable-next-line
+            objName = formatVersionKey(objName, params.versionId);
+        }
+        const m = this.getCollection(METASTORE);
+        m.aggregate([
+            { $match: { _id: bucketName } },
+            {
+                $lookup: {
+                    from: bucketName,
+                    pipeline: [
+                        { $match: { _id: objName } },
+                    ],
+                    as: 'object',
+                },
+            },
+        ], {}, (err, cursor) => {
             if (err) {
                 log.error(
                     'getBucketAttributes: error getting bucket attributes',
```

Comment:
not necessary to start benchmark testing, but if we keep this combined approach, this error message should change